### DISCUSSION TITLE
Specular/Glossiness material to Metal/Rough material conversion

### DISCUSSION THREAD

**Discussion by DanieleMagri** [google/model-viewer#4763] [Q&A]
Hello everyone, I'm having a problem with Three.js no longer supporting specular/glossiness materials. I saw that with glTF Transform it might be possible to convert the material, but I'm having trouble doing it. So, I was wondering if using model viewer this conversion process is automatic?

**Comment by elalish** [comment]
Indeed! It's automatic in our editor: just load the model and it will update it and you can redownload the scene. https://modelviewer.dev/editor/

**Reply by DanieleMagri** [reply]
Actually, I wasn't referring to model viewer as in the editor, but to the component to be integrated into the web app. The 3D models are generated dynamically by a server and I need to receive them, convert them, and use them.

**Reply by elalish** [reply]
We don't build it into MV, but you can follow how we used gltf-transform to convert here: https://github.com/google/model-viewer/blob/master/packages/space-opera/src/components/model_viewer_snippet/components/validation.ts#L186
